package-res/resources/web/dojo/pentaho/common/Dialog.js

```diff
@@ -32,6 +32,10 @@ define(["dojo/_base/declare", "dijit/_WidgetBase", "dijit/_TemplatedMixin", "dij
             closeIcon: undefined,
             _onCancelCallback: undefined,
 
+            constructor: function() {
+              Dialog._DialogLevelManager._beginZIndex = 1051;
```

Comment:
Looks like we can't.
For dijit/Dialog z-index property is hard-coded in js code (950 level). It overrides all css z-index properties.